src/app/helpers/rephraseComment.ts

```diff
@@ -0,0 +1,27 @@
+const { OpenAI } = require('openai');
+const openai = new OpenAI({
+	apiKey: process.env.NEXT_PUBLIC_OPENAI_API_KEY,
```

Comment:
Let's use a [router handler](https://nextjs.org/docs/app/building-your-application/routing/route-handlers) for this as we can't expose our OpenAI API key.

Reply:
For this..do you want to me make a open ai URL call  and pass the the apikey as a Bearer token instead of the above method?

Reply:
No, basically I want you to make a request from client-side Nextjs to a Nextjs API route and from the API route, make the request to the Open AI endpoints. So the API route is acting as a proxy here to protect the API key.

Below is a similar scenario. In this case, you have a weather app and you want to protect your weather service API:

1. First, let's create the API route handler. Create a file called `app/api/weather/route.js`:

```ts
// app/api/weather/route.js

import { NextResponse } from 'next/server';
import axios from 'axios';

const WEATHER_API_KEY = process.env.WEATHER_API_KEY;
const WEATHER_API_URL = 'https://api.weatherapi.com/v1/current.json';

export async function GET(request) {
  const { searchParams } = new URL(request.url);
  const city = searchParams.get('city');

  if (!city) {
    return NextResponse.json({ error: 'City parameter is required' }, { status: 400 });
  }

  try {
    const response = await axios.get(WEATHER_API_URL, {
      params: {
        key: WEATHER_API_KEY,
        q: city,
      },
    });

    return NextResponse.json(response.data);
  } catch (error) {
    return NextResponse.json({ error: 'Error fetching weather data' }, { status: 500 });
  }
}
```

2. Now, let's create a client-side component that uses this API route. Create a file called `components/WeatherWidget.js`:

```javascript
// components/WeatherWidget.js

'use client';

import { useState } from 'react';

export default function WeatherWidget() {
  const [city, setCity] = useState('');
  const [weather, setWeather] = useState(null);

  const fetchWeather = async () => {
    try {
      const response = await fetch(`/api/weather?city=${encodeURIComponent(city)}`);
      const data = await response.json();
      setWeather(data);
    } catch (error) {
      console.error('Error fetching weather:', error);
    }
  };

  return (
    <div>
      <input
        type="text"
        value={city}
        onChange={(e) => setCity(e.target.value)}
        placeholder="Enter city name"
      />
      <button onClick={fetchWeather}>Get Weather</button>
      {weather && (
        <div>
          <h2>Weather in {weather.location.name}</h2>
          <p>Temperature: {weather.current.temp_c}°C</p>
          <p>Condition: {weather.current.condition.text}</p>
        </div>
      )}
    </div>
  );
}
```

This approach still keeps your API key secure on the server side while allowing your client-side code to indirectly use the third-party API.

Reply:
the problem i was facing in this approach was calling  the API internally. when i call "/api/rephrase" it gonna go to "backendurl/api/rephrase" but it needs to go to  "frontend/api/rephrase"

Reply:
You can use axios instead of `axiosInstance`. This way, you can use relative paths:
```ts
import axios from 'axios'

axios.post('/api/rephrase')  // will call your current origin + /api/rephrase
```

Reply:
![image](https://github.com/user-attachments/assets/006ca39d-45b0-40b5-a79d-0ab423c079b5)
![image](https://github.com/user-attachments/assets/cc4e3ad9-ac3e-4dad-a01c-22010220c4fa)

Its goes to backend URL